libs/components/indicators/testing/src/icon/icon-harness.spec.ts

```diff
@@ -0,0 +1,205 @@
+  fixture.componentInstance.variant = variant;
+  fixture.detectChanges();
+  await expectAsync(iconHarness.getVariant()).toBeResolvedTo(variant);
+}
+
+const iconTypes = ['fa', 'skyux'];
+const variants = ['line', 'solid'];
+const sizes = ['lg', '2x', '3x', '4x', '5x'];
+
+describe('Icon harness', () => {
+  async function setupTest(options: { dataSkyId?: string } = {}): Promise<{
+    iconHarness: SkyIconHarness;
+    fixture: ComponentFixture<TestComponent>;
+    loader: HarnessLoader;
+    pageLoader: HarnessLoader;
+  }> {
+    await TestBed.configureTestingModule({
+      declarations: [TestComponent],
+      imports: [SkyIconModule],
+    }).compileComponents();
+
+    const fixture = TestBed.createComponent(TestComponent);
+    const loader = TestbedHarnessEnvironment.loader(fixture);
+    const pageLoader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+
+    const iconHarness: SkyIconHarness = options.dataSkyId
+      ? await loader.getHarness(
+          SkyIconHarness.with({ dataSkyId: options.dataSkyId })
+        )
+      : await loader.getHarness(SkyIconHarness);
+    fixture.detectChanges();
+
+    return { iconHarness, fixture, loader, pageLoader };
+  }
+
+  it('should return the correct icon name for all icon types and variants', async () => {
+    const { iconHarness, fixture } = await setupTest();
+    for (const type of iconTypes) {
+      fixture.componentInstance.iconType = type;
+      for (const variant of variants) {
+        fixture.componentInstance.variant = variant;
+        await validateIconName(
+          iconHarness,
+          fixture,
+          fixture.componentInstance.iconName
+        );
+      }
+    }
+  });
+
+  it('should throw error if icon name is not set', async () => {
+    const { iconHarness, fixture } = await setupTest();
+    fixture.componentInstance.iconName = undefined;
+    await expectAsync(iconHarness.getIconName()).toBeRejectedWithError(
+      'Icon does not exist'
+    );
+  });
+
+  it('should return the correct icon size', async () => {
+    const { iconHarness, fixture } = await setupTest();
+    for (const size of sizes) {
+      await validateIconSize(iconHarness, fixture, size);
+    }
+  });
+
+  it('should return undefined if size is not set', async () => {
+    const { iconHarness, fixture } = await setupTest();
+    await validateIconSize(iconHarness, fixture, undefined);
+  });
+
+  it('should return the correct icon type', async () => {
+    const { iconHarness, fixture } = await setupTest();
+    for (const type of iconTypes) {
+      await validateIconType(iconHarness, fixture, type);
+    }
+  });
+
+  it('should return the default icon type', async () => {
+    const { iconHarness, fixture } = await setupTest();
+    fixture.componentInstance.iconType = undefined;
+    fixture.detectChanges();
+    await expectAsync(iconHarness.getIconType()).toBeResolvedTo('fa');
+  });
+
+  it('should return the right variant for skyux icons', async () => {
```

Comment:
Do you mean "correct value"? I would avoid the word "right" since that word is used often in JavaScript for offset values.